### PR TITLE
Fixes #18981: Slow computation of dynamic groups on large system

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/18981

### CODE REVIEW

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala [hunk 1]
```diff
@@ -441,6 +445,7 @@ class LDAPBasedConfigService(
        rudder.generation.delay=0s
        rudder.generation.trigger=all
        node.accept.duplicated.hostname=false
+       rudder.compute.dyngroups.max.parallelism=1
```
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala:448]: I think that for consistancy sack, it should be like `rudder.generation.max.parallelism`, ie accepting either an integer like here, or `xF`, with `F` a float and `x` the character `x`, meaning `F time the number of CPU available`. The same parsing logic can be used (and can even be factored out in an utility object, which would also ensure to always return an integer at least equals to `1`).
And with that, setting it to `x0.5` (ie, `number CPU / 2`) seems to be a good default.

**Review by fanf — COMMENTED** [review]
Appart for some little enhancement in comments, this seems to be a good PR with limited risk and clear gain. Nice !

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala (hunk 1 above)
Comment (ncharles) [reply webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala:448]: it's not really generation, as it is to compute dynamic groups (so also done outside of policy generation)

i'll change the xF

Comment (ncharles) [reply webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/appconfig/ConfigService.scala:448]: I'm not sure i'd like to change the default to something larger than 1, as it depends on the size of the LDAP connection pool which is 2 by default

**Comment by ncharles** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/batch/UpdateDynamicGroups.scala [hunk 2]
```diff
@@ -227,16 +229,17 @@ class UpdateDynamicGroups(
...
-              error.rootExceptionCause.foreach(ex =>
-                logger.error("Exception was:", ex)
-              )
-            case Full(diff) =>
+          eitherRes match {
+            case Left(e) =>
+              val error = (e.fullMsg + s"Error when updating dynamic group '${id.value}'")
+              logger.error(error)
+             // e.fullMsg.foreach(ex =>
+             //   logger.error("Exception was:", ex)
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/batch/UpdateDynamicGroups.scala:240]: this comment should be removed

**Comment by ncharles** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**